pkgs/tools/compression/xz/xz-after-bootstrap.nix

```
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  automake,
+  autoconf,
+  perl,
+  libtool,
+  gettext,
+  perl538Packages,
+  xz,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "xz";
+  version = xz.version;
+
+  src = fetchFromGitHub {
+    owner = "tukaani-project";
+    repo = "xz";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-Xp1uLtQIoOG/qVLpq5D/KFmTOJ0+mNkNclyuJsvlUbE=";
+  };
+
+  strictDeps = true;
+
+  outputs = [
+    "bin"
+    "dev"
+    "out"
+    "man"
+    "doc"
+  ];
+
+  enableParallelBuilding = true;
+
+  doCheck = true;
+
+  nativeBuildInputs = [
+    gettext
+    libtool
+    automake
+    autoconf
+    perl
+    perl538Packages.Po4a
+  ];
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  postPhases = [ "compareArtifacts" ];
+
+  compareArtifacts = ''
+    diff $out/lib/liblzma.so ${xz.out}/lib/liblzma.so
```

```suggestion
    diff $out/lib/liblzma.so ${xz.out}/lib/liblzma${lib.optionalString stdenv.hostPlatform.hasSharedLibraries stdenv.hostPlatform.extensions.sharedLibrary}
```
Should fix building on Darwin